platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/CommonPcesWriter.java

```diff
@@ -0,0 +1,414 @@
+
+    /**
+     * Close the output file.
+     * <p>
+     * Should only be called if {@link #currentMutableFile} is not null.
+     */
+    private void closeFile() {
+        try {
+            previousSpan = currentMutableFile.getUtilizedSpan();
+            if (!bootstrapMode) {
+                averageSpanUtilization.add(previousSpan);
+            }
+            currentMutableFile.close();
+            lastFlushedEvent = lastWrittenEvent;
+
+            fileManager.finishedWritingFile(currentMutableFile);
+            currentMutableFile = null;
+
+            // Not strictly required here, but not a bad place to ensure we delete
+            // files incrementally (as opposed to deleting a bunch of files all at once).
+            pruneOldFiles();
+        } catch (final IOException e) {
+            throw new UncheckedIOException("unable to prune files", e);
+        }
+    }
+
+    /**
+     * Delete old files from the disk.
+     */
+    private void pruneOldFiles() {
+        if (!streamingNewEvents) {
+            // Don't attempt to prune files until we are done replaying the event stream (at start up).
+            // Files are being iterated on a different thread, and it isn't thread safe to prune files
+            // while they are being iterated.
+            return;
+        }
+
+        try {
+            fileManager.pruneOldFiles(minimumAncientIdentifierToStore);
+        } catch (final IOException e) {
+            throw new UncheckedIOException("unable to prune old files", e);
+        }
+    }
+
+    /**
+     * Prepare the output stream for a particular event. May create a new file/stream if needed.
+     *
+     * @param eventToWrite the event that is about to be written
+     * @return true if this method call resulted in the current file being closed
+     */
+    public boolean prepareOutputStream(@NonNull final PlatformEvent eventToWrite) throws IOException {
+        boolean fileClosed = false;
+        if (currentMutableFile != null) {
+            final boolean fileCanContainEvent =
+                    currentMutableFile.canContain(eventToWrite.getAncientIndicator(fileType));
+            final boolean fileIsFull =
+                    UNIT_BYTES.convertTo(currentMutableFile.fileSize(), UNIT_MEGABYTES) >= preferredFileSizeMegabytes;
+
+            if (!fileCanContainEvent || fileIsFull) {
+                closeFile();
+                fileClosed = true;
+            }
+
+            if (fileIsFull) {
+                bootstrapMode = false;
+            }
+        }
+
+        // if the block above closed the file, then we need to create a new one
+        if (currentMutableFile == null) {
+            final long upperBound = nonAncientBoundary
+                    + computeNewFileSpan(nonAncientBoundary, eventToWrite.getAncientIndicator(fileType));
+
+            currentMutableFile = fileManager
+                    .getNextFileDescriptor(nonAncientBoundary, upperBound)
+                    .getMutableFile();
```

Comment:
The main feature of the inline PCES writer should be that it ensures that events are persisted before returning. The way this is currently implemented is by calling `getMutableFile(false, true)`

Reply:
added to the `CommonPcesWriter`